hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4020,6 +4020,17 @@ func (r *HostedClusterReconciler) validateHostedClusterSupport(hc *hyperv1.Hoste
 func (r *HostedClusterReconciler) validateNetworks(hc *hyperv1.HostedCluster) error {
 	errs := validateSliceNetworkCIDRs(hc)
 
+	ip := getNodePortIP(hc)
+	if ip != nil {
+		// Validate that the nodeport IP is not within the ServiceNetwork CIDR.
+		for _, cidr := range hc.Spec.Networking.ServiceNetwork {
+			netCIDR := (net.IPNet)(cidr.CIDR)
+			if netCIDR.Contains(ip) {
+				errs = append(errs, field.Invalid(field.NewPath("spec.networking.ServiceNetwork"), cidr.CIDR.String(), fmt.Sprintf("Nodeport IP is within the service network range: %s is within %s", ip, cidr.CIDR.String())))
+			}
+		}
+	}
+
```

Comment:
I would add this to a separated function to keep this validateNetwork clean to append different validations, this way is more clear and readable.